### PR TITLE
[android/CHIPTool] Add missing parts for rendezvous over BLE

### PR DESCRIPTION
**Pull request by Damian-Nordic**
#### Problem
Android CHIPTool has incomplete implementation of rendezvous over BLE.

 #### Summary of Changes
* Pass Bluetooth Gatt events to CHIP BleLayer
* Fix RendezvousSession in case a connection object is provided in RendezvousParams instead of discriminator
* Call beginConnectBle() in "Rendezvous over BLE" button handler to initiate secure connection to the device

 This is another major step to implement #2715, but some kind of network provisioning needs to be added, too.

### CODE REVIEW

File: src/controller/java/src/chip/devicecontroller/AndroidChipStack.java [hunk 1]
```diff
@@ -65,6 +65,98 @@
     }
   }
 
+  public static class ChipGattCallback extends BluetoothGattCallback {
+    @Override
+    public void onConnectionStateChange(BluetoothGatt gatt, int status, int newState) {
+      if (newState == BluetoothProfile.STATE_DISCONNECTED) {
```
Comment (andy31415) [inline src/controller/java/src/chip/devicecontroller/AndroidChipStack.java:71]: nit: you could decrease indent by switching condition and early-returning here and below:

```
if (newState != BluetoohProfile.STATE_DISCONNECTED) {
  return;
}

int connId = getInstance().getConnId(gatt);
...
```

**Review by andy31415 — APPROVED** [review]

File: src/controller/java/src/chip/devicecontroller/AndroidChipStack.java (hunk 1 above)
Comment (Damian-Nordic) [reply src/controller/java/src/chip/devicecontroller/AndroidChipStack.java:71]: right, I'll change it

**Review by BroderickCarlin — APPROVED** [review]

**Review by rwalker-apple — APPROVED** [review]

**Comment by BroderickCarlin** [comment]
@Damian-Nordic can you fix the merge conflict?

**Comment by Damian-Nordic** [comment]
> @Damian-Nordic can you fix the merge conflict?

Sure, done!

**Pull request merged**